### ISSUE TITLE
VueJS TypeScript Property Decorators Error

### ISSUE THREAD

**Issue by slavarazum** [storybookjs/storybook#10327]
**Describe the bug**
Looks like `@storybook/preset-typescript` can't interpret decorators from `vue-property-decorators`, like: `@Component`

**Error screenshot**:
![screenshot](https://image.prntscr.com/image/6Eig_CO4SPe7hiC57Ewsxg.png)

Error text:
> Support for the experimental syntax 'decorators-legacy' isn't currently enabled

**Code snippets**

_main.js_
```js
const path = require('path');

// your app's webpack.config.js
const custom = require('../webpack.config.js');

module.exports = {
  stories: ['../stories/**/*.stories.[tj]s'],
  addons: [
    '@storybook/preset-typescript',
    '@storybook/addon-actions',
    '@storybook/addon-links',
  ],

  webpackFinal: (config) => {
    return {
      ...config,
      resolve: {
        ...config.resolve,
        alias: custom.resolve.alias,
      }
    };
  }
};
```

_1-TestComponent.stories.ts_

```js
import { action } from '@storybook/addon-actions';

import TestComponent from './TestComponent';

export default {
  title: 'TestComponent',
  component: TestComponent,
};

export const Simple = () => ({
  components: { TestComponent },
  template: '<TestComponent>Some text</TestComponent>',
  methods: { action: action('clicked') },
});
```

_TestComponent.vue_
```js
<template>
  <div>
    <slot />
  </div>
</template>
<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component({ name: 'TestComponent' })
export default class TestComponent extends Vue {}
</script>
```

**System:**
```
Environment Info:

  System:
    OS: macOS 10.15.4
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
  Binaries:
    Node: 13.10.1 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.7 - /usr/local/bin/npm
  Browsers:
    Chrome: 80.0.3987.149
    Firefox: 74.0.1
    Safari: 13.1
  npmPackages:
    @storybook/addon-actions: ^5.3.18 => 5.3.18 
    @storybook/addon-links: ^5.3.18 => 5.3.18 
    @storybook/addons: ^5.3.18 => 5.3.18 
    @storybook/preset-typescript: ^3.0.0 => 3.0.0 
    @storybook/vue: ^5.3.18 => 5.3.18 
```

**Additional context**
Also have tried to pass `tsconfig.json` path:

_main.js_

```js
addons: [
    {
      name: '@storybook/preset-typescript',
      options: {
        tsLoaderOptions: {
          configFile: path.resolve(__dirname, '../tsconfig.json'),
        },
      },
    },
  ],
```

**Comment by metanas** [comment]
You should add `ts-loader` to your webpack configuration in main.js of your storybook
```js
module.exports = {
  stories: ['../stories/**/*.stories.[tj]s'],
  addons: [
    '@storybook/preset-typescript',
    '@storybook/addon-actions',
    '@storybook/addon-links',
  ],

  webpackFinal: (config) => {
    config.module.rules.push({
      test: /\.ts$/,
      exclude: /node_modules/,
      use: [{
        loader: 'ts-loader',
        options: {
          appendTsSuffixTo: [/\.vue$/],
          transpileOnly: true
        },
      }],
    });
    return {
      ...config,
      resolve: {
        ...config.resolve,
        alias: custom.resolve.alias,
      }
    };
  }
};

**Comment by shilman** [comment]
Thanks for reporting @slavarazum and for the workaround @metanas. We have an open PR for that here, but are still figuring out the best path forward: https://github.com/storybookjs/presets/pull/122

**Comment by slavarazum** [comment]
@metanas Thanks for help! I thought `ts-loader` included under the hood.

Also, for non vue-clie presets we should add runtime compiler:

```js
config.resolve.alias['vue$'] = 'vue/dist/vue.esm.js';
```

@shilman Thanks for the link, just have subscribed.

**Comment by shilman** [comment]
We've just released zero-config typescript support in 6.0-beta. Please upgrade and test it!

- [Release 6.0.0-beta.12+](https://github.com/storybookjs/storybook/releases/tag/v6.0.0-beta.12)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#zero-config-typescript)
- [Updated docs](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/configurations/typescript-config/index.md)

Thanks for your help and support getting this stable for release!

**Comment by caryuval** [comment]
Before messing with the TS configs or upgrading Storybook, I suggest making sure that you didn't forget to add 
`lang="ts"` in your script tag, like so: `<script lang="ts">`
